azext_edge/edge/providers/check/dataflow.py

```diff
@@ -1565,3 +1724,119 @@ def evaluate_dataflow_profiles(
+):
+    check_manager = CheckManager(
+        check_name=registry_endpoint_check_name,
+        check_desc=registry_endpoint_check_desc,
+    )
+    target = registry_endpoint_target
+
+    all_registry_endpoints = get_resources_by_name(
+        api_info=DATAFLOW_ACTIVE_API,
+        kind=DataflowResourceKinds.REGISTRYENDPOINT,
+        resource_name=resource_name,
+    )
+    if not all_registry_endpoints:
+        no_registry_endpoints_text = "No Registry Endpoints detected in any namespace."
+        check_manager.add_target(target_name=target)
+        check_manager.add_target_eval(
+            target_name=target,
+            status=CheckTaskStatus.skipped.value,
+            value={"registryEndpoints": no_registry_endpoints_text},
+        )
+        check_manager.add_display(
+            target_name=target,
+            display=Padding(no_registry_endpoints_text, (0, 0, 0, PADDING)),
+        )
+        return check_manager.as_dict(as_list=as_list)
+
+    for namespace, registry_endpoints in get_resources_grouped_by_namespace(all_registry_endpoints):
+        check_manager.add_target(
+            target_name=target,
+            namespace=namespace,
+            conditions=[
+                "endsWith(spec.host, 'azurecr.io')",
+                "spec.authentication.method",
+            ],
+        )
+        check_manager.add_display(
+            target_name=target,
+            namespace=namespace,
+            display=Padding(
+                f"Registry Endpoints in namespace {{[purple]{namespace}[/purple]}}",
+                (0, 0, 0, PADDING),
+            ),
+        )
+
+        for registry_endpoint in list(registry_endpoints):
+            registry_endpoint_name = registry_endpoint.get("metadata", {}).get("name")
+            spec = registry_endpoint.get("spec", {})
+            host = spec.get("host", "")
+
+            check_manager.add_display(
+                target_name=target,
+                namespace=namespace,
+                display=Padding(
+                    f"\n- Registry Endpoint {{{colorize_string(value=registry_endpoint_name)}}} {colorize_string(color='green', value='detected')}",
+                    (0, 0, 0, PADDING),
+                ),
+            )
+
+            # evaluate status
+            status = registry_endpoint.get("status", {})
+            _process_dataflow_resource_status(
+                check_manager=check_manager,
+                target_name=target,
+                namespace=namespace,
+                status=status,
+                resource_name=registry_endpoint_name,
+                resource_kind=DataflowResourceKinds.REGISTRYENDPOINT.value,
+                detail_level=detail_level,
+                padding=INNER_PADDING,
+            )
+
+            # evaluate host condition - endsWith(spec.host, 'azurecr.io')
+            host_status = CheckTaskStatus.success if host.endswith("azurecr.io") else CheckTaskStatus.error
```

Comment:
does casing matter with azurecr.io ?

Reply:
confirmed, it does - essentially case-sensitive `*.azurecr.io`